crates/kubelet/src/kubelet.rs

```diff
@@ -255,16 +256,18 @@ mod test {
     struct PodState;
 
     #[async_trait::async_trait]
-    impl AsyncDrop for PodState {
-        type ProviderState = ProviderState;
+    impl ResourceState for PodState {
+        type Manifest = Pod;
+        type Status = Status;
+        type SharedState = ProviderState;
```

Comment:
This is an elegant solution to the `async_drop` problem and I like it **much** better than mine - thank you!